vnu-java-resolver.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+const { spawnSync, execSync } = require('child_process');
+const { mkdirSync, existsSync, createWriteStream } = require('fs');
+const { join } = require('path');
+const https = require('https');
+const { extract } = require('tar');
+
+const CACHE_DIR = join(__dirname, '..', 'node_modules', '.cache', 'vnu-jar', 'java');
+const TEMURIN_BASE_URL = 'https://github.com/adoptium/temurin17-binaries/releases/latest/download/';
```

Comment:
Should we consider locking to particular versions? In the event the latest release doesn't include a particular platform/arch build?